### PR TITLE
Checklist: Hide banner from too old sites.

### PR DESCRIPTION
**Pull request by taggon**
The onboarding checklist may not work properly for the sites created before the feature was launched due to lack of `onboarding_checklist_initialized` option that is generated when a site is created. This PR will check if the site is eligible for the checklist and hide the banner if not.

_Note_: You need to apply D17548-code to your sandbox first.

## Test plan

1. Create a new site.
2. See if the checklist banner shows up on the stats page. You should be able to see it.
3. Repeat the above step on the site created before Feb 1, 2018. But this time, the banner should not appear on the page.

### CODE REVIEW

**Comment by taggon** [comment]
Rebased

File: client/state/sites/constants.js [hunk 1]
```diff
@@ -38,9 +38,10 @@ export const SITE_REQUEST_OPTIONS = [
 	'is_mapped_domain',
 	'is_redirect',
 	'is_wpcom_store',
-  'woocommerce_is_active',
+	'woocommerce_is_active',
 	'jetpack_version',
 	'main_network_site',
+	'onboarding_checklist_initialized',
```
Comment (gwwar) [inline client/state/sites/constants.js]: Do we really need this off the site object? Please be careful about bloating the site object further. We do have the option of adding additional endpoints if necessary.

Do you mind expanding on what this prop means? It still looks like the client is performing a number of checks in addition to looking at this prop.

**Comment by damiannep** [comment]
@taggon Is this also related to https://github.com/Automattic/wp-calypso/issues/26890 or does it solve it?

File: client/state/sites/constants.js (hunk 1 above)
Comment (mattwiebe) [reply client/state/sites/constants.js]: It basically means that the checklist has been initialized on the server side. I think that in practice it should be pretty redundant with the site age check already happening in the reducer so I'll just eliminate it as I'd like to get this out the door.

**Review by mattwiebe — APPROVED** [review]

**Comment by mattwiebe** [comment]
@damiannep this will solve that issue

File: client/state/sites/constants.js (hunk 1 above)
Comment (gwwar) [reply client/state/sites/constants.js]: Thanks for the extra context @mattwiebe !

**Comment by dmsnell** [comment]
Did we explore performing this login in the backend? Do we not already poll for the checklist for a site? If the site is ineligible or too old what would prevent us from simply returning a response that indicates as much?

We could keep this code out of Calypso which would help guard against bloat and we could make sure that the mobile teams wouldn't have to recreate the logic and keep it in sync on their end.

**Pull request merged**